Generals/Code/GameEngine/Source/Common/System/UnicodeString.cpp

```diff
@@ -94,10 +94,8 @@ void UnicodeString::ensureUniqueBufferOfSize(int numCharsNeeded, Bool preserveDa
 		return;
 	}
 
+	DEBUG_ASSERTCRASH(numCharsNeeded <= MAX_LEN, ("UnicodeString::ensureUniqueBufferOfSize exceeds max string length %d with requested length %d", MAX_LEN, numCharsNeeded));
 	int minBytes = sizeof(UnicodeStringData) + numCharsNeeded*sizeof(WideChar);
-	if (minBytes > MAX_LEN)
```

Comment:
This did not work as advertised.
```
MAX_LEN = 32767							///< max total len of any UnicodeString, in chars
```
But minBytes was in bytes + header, not number of characters. Bad.